x-pack/platform/plugins/shared/observability_ai_assistant/server/plugin.ts

```diff
@@ -130,23 +130,19 @@ export class ObservabilityAIAssistantPlugin
     }));
 
     // Update existing index assets (mappings, templates, etc). This will not create assets if they do not exist.
-    const indexAssetsUpdatedPromise = updateExistingIndexAssets({
-      logger: this.logger.get('index_assets'),
-      core,
-    }).catch((e) => this.logger.error(`Index assets could not be updated: ${e.message}`));
-
-    // register task to migrate knowledge base entries to include semantic_text field
-    registerAndScheduleKbSemanticTextMigrationTask({
-      core,
-      taskManager: plugins.taskManager,
-      logger: this.logger.get('kb_semantic_text_migration_task'),
-      config: this.config,
-      indexAssetsUpdatedPromise,
-    }).catch((e) =>
-      this.logger.error(
-        `Knowledge base semantic_text migration task could not be registered: ${e.message}`
+    updateExistingIndexAssets({ logger: this.logger, core })
+      .then(() =>
+        populateMissingSemanticTextFieldMigration({
+          core,
+          logger: this.logger,
+          config: this.config,
+        })
       )
-    );
+      .catch((e) =>
```

Comment:
Does `createOrUpdateIndexAssets` need to throw in order for this error to be caught or prevent `populateMissingSemanticTextFieldMigration` from running if that's the intention?